### DISCUSSION TITLE
how to run separate jupyter remote server for each notebook in a devcontainer?

### DISCUSSION THREAD

**Discussion by mkarikom** [microsoft/vscode-jupyter#11227] [Questions and Answers]
### Background:
Sometimes it's nice to be able to explore two different versions of some code.  In the notebook context (at least in vscode), this seems to require separate kernels.  My understanding is that (at least in vscode), having two separate kernels requires two separate servers.  For kernel persistence (remote connection here can be dodgy), these two servers are started as "remote servers" as below.

#### Note: this scheme already works great for the case with a single notebook/ jupyter server 
For each of two notebooks, I open up a screen session and run the following (each kernel has a unique port):
1.  create a server for notebook 1: `jupyter notebook --no-browser --port=9996 --config=/workspaces/working_repo/.devcontainer/jupyter_notebook_config.json`
2. create a server for notebook 2: `jupyter notebook --no-browser --port=9997 --config=/workspaces/working_repo/.devcontainer/jupyter_notebook_config.json`
3. open notebook 1 and select the jupyter server on port 9996.
4. open notebook 2 and select the jupyter server on port 9997
5. go back to notebook 1 and notice that the server has been overridden (it now tries to use the one on port 9996)

**Comment by DonJayamanne** [comment]
I've transferred this into an issue, we currently do not support such a feature, please follow the https://github.com/microsoft/vscode-jupyter/issues/11266 issue for further updates.
